src/github.com/matrix-org/dendrite/clientapi/storage/syncserver.go

```diff
@@ -81,6 +85,16 @@ func (d *SyncServerDatabase) SetPartitionOffset(topic string, partition int32, o
 	return d.partitions.UpsertPartitionOffset(topic, partition, offset)
 }
 
+// SyncStreamPosition returns the latest position in the sync stream
+func (d *SyncServerDatabase) SyncStreamPosition() (int64, error) {
```

Comment:
Yes, it should, but that's going to be painful to do because it'll introduce a circular loop in the `import` graph which Go will die on. I'll need to move it out to a 3rd package. I was going to punt that for another PR.